src/rapids_pre_commit_hooks/alpha_spec.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import reduce
+
+import yaml
+from packaging.requirements import Requirement
+from packaging.specifiers import SpecifierSet
+
+from .lint import LintMain
+
+RAPIDS_ALPHA_SPEC_PACKAGES = {
```

Comment:
I don't think `rapids-build-backend` belongs here, since it doesn't have a CUDA-suffixed name, depend on any CUDA-suffixed packages, or depend on any alpha spec packages.

Let me look into `dask-cuda`.

Reply:
Historically we used pre-releases on PyPI: https://pypi.org/project/dask-cuda/#history

It looks like we haven't done that since 24.02, and nightlies are instead hosted here: https://anaconda.org/rapidsai-nightly/dask-cuda

Reply:
Looks like projects that depend on `dask-cuda` do indeed use an alpha spec: https://github.com/rapidsai/cudf/blob/27220d6aba0728930f6e2922a7a27e9e503c2add/ci/build_wheel.sh#L39

I'll go ahead and add it

Reply:
Yeah, dask-cuda was switched over to the same versioning scheme as the rest of RAPIDS at the same time that we switched indexes. [The lists I had in RBB before we switched it to use dfg should be a decent guide](https://github.com/rapidsai/rapids-build-backend/blob/8cfa39b83a0bda447eaf11712e50a6f817f659cc/rapids_build_backend/impls.py#L95).